kaldb/src/main/java/com/slack/kaldb/chunk/SerialS3ChunkDownloaderImpl.java

```diff
@@ -0,0 +1,30 @@
+package com.slack.kaldb.chunk;
+
+import static com.slack.kaldb.logstore.BlobFsUtils.copyFromS3;
+
+import com.slack.kaldb.blobfs.BlobFs;
+import java.nio.file.Path;
+
+/**
+ * SerialS3ChunkDownloaderImpl downloads all the chunk related data from S3 to a local store one
+ * file at a time.
+ */
+public class SerialS3ChunkDownloaderImpl implements ChunkDownloader {
+  private final String s3Bucket;
+  private final String snapshotId;
+  private BlobFs blobFs;
+  private Path dataDirectory;
+
+  public SerialS3ChunkDownloaderImpl(
+      String s3Bucket, String snapshotId, BlobFs blobFs, Path localDataDirectory) {
+    this.s3Bucket = s3Bucket;
+    this.snapshotId = snapshotId;
+    this.blobFs = blobFs;
+    this.dataDirectory = localDataDirectory;
+  }
+
+  @Override
+  public boolean download() throws Exception {
+    return copyFromS3(s3Bucket, snapshotId, blobFs, dataDirectory).length == 0;
```

Comment:
how would this API work for rate limiting? Wouldn't it need to be done on a per file basis?

Reply:
Currently, the plan is to control the number of chunks downloaded in parallel. For that, I am adding a wrapper around this downloader that will limit the number of chunks downloaded in parallel in a different PR. Since, we only have a serial s3 file downloader, this will work for adding the rate limits.